src/OpenRiaServices.Tools/Framework/SharedCodeGenUtilities.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace OpenRiaServices.Tools
+{
+    /// <summary>
+    /// A collection of code generation utilities.
+    /// </summary>
+    internal static partial class SharedCodeGenUtilities
+    {
+        public static EndpointRoutePattern? TryGetRoutePatternFromAssembly(Assembly assembly)
+        {
+#if NET
+            if(assembly?.CustomAttributes.FirstOrDefault(cad => cad.AttributeType.FullName ==  "OpenRiaServices.Hosting.AspNetCore.DomainServiceEndpointRoutePatternAttribute")
+                        is { ConstructorArguments: [var constructorArgument] } )
+            {
+                return (EndpointRoutePattern)Convert.ToInt32(constructorArgument.Value, CultureInfo.InvariantCulture);
```

Comment:
This a boxed int, so could change  to
```suggestion
                return (EndpointRoutePattern)(int)constructorArgument.Value;
```